pkg/controllers/controller.go

```diff
@@ -49,18 +50,22 @@ func (c *GenericController) Reconcile(ctx context.Context, req reconcile.Request
 	}
 	// 2. Copy object for merge patch base
 	persisted := resource.DeepCopyObject()
-	// 3. Set to true to remove race condition
+	// 3. Set Defaults
+	if _, ok := resource.(apis.Defaultable); ok {
+		resource.(apis.Defaultable).SetDefaults(ctx)
+	}
+	// 4. Set to true to remove race condition
 	// TODO: remove status conditions on provisioners
 	if provisioner, ok := resource.(*v1alpha1.Provisioner); ok {
```

Comment:
Similarly, can you use the condition manager interface while you're in here? This would remove any special provisioner logic in this file.